warp10/src/main/java/io/warp10/script/functions/ASENCODER.java

```diff
@@ -0,0 +1,89 @@
+//
+//   Copyright 2019  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.functions;
+
+import com.google.common.base.Charsets;
+import io.warp10.continuum.gts.GTSEncoder;
+import io.warp10.continuum.gts.GTSWrapperHelper;
+import io.warp10.continuum.gts.GeoTimeSerie;
+import io.warp10.continuum.store.thrift.data.GTSWrapper;
+import io.warp10.crypto.OrderPreservingBase64;
+import io.warp10.script.ElementOrListStackFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptStack;
+import org.apache.thrift.TDeserializer;
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TCompactProtocol;
+
+import java.io.IOException;
+
+/**
+ * Create a GTSEncoder for each instance of GTS, wrapped GTS or raw wrapped GTS.
+ */
+public class ASENCODER extends ElementOrListStackFunction {
+
+  private final ElementStackFunction gtstoFunction = new ElementStackFunction() {
+    @Override
+    public Object applyOnElement(Object element) throws WarpScriptException {
+      // If wrap, convert to raw wrap
+      if (element instanceof String) {
+        element = OrderPreservingBase64.decode(element.toString().getBytes(Charsets.US_ASCII));
+      }
+
+      // If raw wrap, convert to GTS
+      if (element instanceof byte[]) {
+        TDeserializer deser = new TDeserializer(new TCompactProtocol.Factory());
+
+        try {
+          GTSWrapper wrapper = new GTSWrapper();
+
+          deser.deserialize(wrapper, (byte[]) element);
+
+          element = GTSWrapperHelper.fromGTSWrapperToGTS(wrapper);
```

Comment:
It will also allow a wrapped encoder to be given to `ASENCODERS` without error if it contains multiple types. Not sure if it is useful but it is logical.